runtime-builder/go-build.sh

```diff
@@ -29,18 +29,45 @@ fi
 export PATH=/usr/local/go/bin:"${PATH}"
 export GOPATH="${workspace}"/_gopath
 
-# Move application files into a temporary staging directory, dependencies excluded.
-staging=$(mktemp -d staging.XXXX)
-find . -mindepth 1 -maxdepth 1 ! -name "${staging}" ! -name "$(basename $GOPATH)" -exec mv {} "${staging}"/ \;
+# $workspace contains files and directories in original app.yaml directory. It
+# consists of the main package files as well as application resource files.
+# If there are dependency packages, $workspace will also contain _gopath
+# directory containing these with the proper GOPATH structure.
+
+
+# If there is _gopath/main-package-path file, reconstruct the main path relative
+# to GOPATH/src by moving files over.
+mainpath=""
+if [[ -f "${GOPATH}/main-package-path" ]]; then
+  mainpkg=$(cat ${GOPATH}/main-package-path)
+  if [[ "$mainpkg" != "" ]]; then
```

Comment:
-n "$mainpkg" ? I don't know if it makes any difference.

Reply:
I don't think it makes any difference.  -n is for testing against null value, which does somehow test true if empty string as well.  I could be wrong on this, but since I'm assigning mainpkg to some value, seems more appropriate to check for empty string instead of null.